File: docs/src/testers/functions/management/version_management.ts
// @sniptest filename=version_management.ts
// @sniptest show=5-16
import assert from 'node:assert/strict';

const functionId = process.env.NOTTE_FUNCTION_ID!;

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

const fn = client.NotteFunction({ function_id: functionId });
const response = await fn.get();

// Get all versions
console.log('Available versions:', response.versions);

// Get latest version
console.log(`Latest: ${response.latest_version}`);

assert.equal(response.function_id, functionId);
const results = [response.versions.length > 0, response.versions.includes(response.latest_version)];
export { results };

File: docs/src/testers/functions/management/version_management.py
# @sniptest filename=version_management.py
# @sniptest show=6-16
# ruff: noqa: E402 -- hidden live fixture precedes displayed imports
import os

function_id = os.environ["NOTTE_FUNCTION_ID"]

from notte_sdk import NotteClient

client = NotteClient()

function = client.Function(function_id=function_id)

# Get all versions
print(f"Available versions: {function.response.versions}")

# Get latest version
print(f"Latest: {function.response.latest_version}")

assert function.response.function_id == function_id
results = [bool(function.response.versions), function.response.latest_version in function.response.versions]
